src/gap-skipper.js

```diff
@@ -0,0 +1,85 @@
+import Ranges from './ranges';
+
+let seeking = false;
+let consecutiveUpdates = 0;
+let timer = null;
+let playerState;
+let lastRecordedTime;
+let adaptiveSeeking;
+let player;
+let tech;
+let gapSkipper;
+
+gapSkipper = function(options) {
+  player = this;
+  tech = options.tech;
+
+  // Allows us to mimic a waiting event in chrome
+  player.on('timeupdate', function() {
+    if (player.paused()) {
+      return;
+    }
+    let currentTime = player.currentTime();
+
+    if (consecutiveUpdates === 5 && currentTime === lastRecordedTime) {
+      // trigger waiting
+      if (playerState !== 'waiting') {
+        consecutiveUpdates = 0;
+        playerState = 'waiting';
+        tech.trigger('adaptive-seeking');
+      }
+    } else if (currentTime === lastRecordedTime) {
+      consecutiveUpdates++;
+    } else {
+      consecutiveUpdates = 0;
+      lastRecordedTime = currentTime;
+    }
+  });
+
+  // Don't listen for waiting while seeking
+  player.on('seeking', function() {
+    seeking = true;
+  });
+
+  // Listen for waiting when finished seeking
+  player.on('seeked', function() {
+    seeking = false;
+  });
+
+  tech.on('playing', function() {
+    tech.on('waiting', adaptiveSeeking);
+  });
+
+  tech.on('adaptive-seeking', adaptiveSeeking);
+
+  tech.on('error', function() {
```

Comment:
player should fire this event as well so that we can remove tech usage in gap-skipper